llvm/include/llvm/CodeGen/SelectionDAGNodes.h

```diff
@@ -452,6 +452,13 @@ struct SDNodeFlags {
   bool hasNoFPExcept() const { return NoFPExcept; }
   bool hasUnpredictable() const { return Unpredictable; }
 
+  bool hasAny() const {
+    return NoUnsignedWrap || NoSignedWrap || Exact || Disjoint || NonNeg ||
+           NoNaNs || NoInfs || NoSignedZeros || AllowReciprocal ||
+           AllowContract || ApproximateFuncs || AllowReassociation ||
```

Comment:
I thought of the FMFs, only NoNaNs, NoInfs, and NoSignedZeros are poison generating.

Reply:
That's my understanding as well.

Reply:
Dropped the ones which obviously correspond to IR level flags.  Any clue on NoFPExcept or Unpredictable?  As a guess, probably not poison generating?

Reply:
Unpredictable is definitely not poison-generating.

NoFPExcept I'm not sure about, but I'd assume that one results in UB rather than poison.

Reply:
Closest I could find to a definition on NoFPExcept is from the constrained intrinsics (https://llvm.org/docs/LangRef.html#id2340), and I agree that appears non-poison generating.  (Except maybe on FP flags?  But I don't think we model it that way.)  At the very least, the ValueTracking code doesn't seem to check for this property on constrained intrinsics.